src/core/analyzer.ts

```diff
@@ -203,17 +206,10 @@ export class Analyzer {
     return (await this._analysisComplete).load(resolvedUrl);
   }
 
-  /**
-   * Returns `true` if the given `url` can be resolved.
-   */
-  canResolveUrl(url: string): boolean {
-    return this._urlResolver.canResolve(this.brandUserInputUrl(url));
-  }
-
   /**
    * Resoves `url` to a new location.
    */
-  resolveUrl(url: string): ResolvedUrl {
+  resolveUrl(url: string): ResolvedUrl|undefined {
```

Comment:
Why not return the original URL if it can't be resolved like in AnalysisContext? I know that would be a change in behavior, but it seems worth it to be consistent. If you want to keep this behavior, maybe change the name of one the methods to indicate the difference in behaviour?

If it's a common pattern to fallback to the input in case the URL can't be resolved, maybe we want an additional method on UrlResolver to do that - it could be implemented in the base class.

Reply:
We don't want to fall back to the original URL, the client should do that if that's the behavior it wants (which most of them don't).